src/firestore/delete.ts

```diff
@@ -248,23 +274,17 @@ export class FirestoreDelete {
       body = this.collectionDescendantsQuery(allDescendants, batchSize, startAfter);
     }
 
-    return api
-      .request("POST", "/v1beta1/" + url, {
-        auth: true,
-        data: body,
-        origin: api.firestoreOriginOrEmulator,
-      })
-      .then((res) => {
-        // Return the 'document' property for each element in the response,
-        // where it exists.
-        return res.body
-          .filter((x: any) => {
-            return x.document;
-          })
-          .map((x: any) => {
-            return x.document;
-          });
-      });
+    return this.apiClient.post("/v1/" + url, body).then((res) => {
+      // Return the 'document' property for each element in the response,
+      // where it exists.
+      return (res.body as any[])
```

Comment:
The `post` method allows for templating so you don't have to do this. The first is the request body type, the second is the response body type (maybe this is too specific? attempting to show an array of objects w/ documents):

```typescript
this.apiClient.post<any, Array<{ document: any }>>(...)
```

Reply:
Nice!